test/Server/SocketTest/SaveUploadTests.cs

```diff
@@ -0,0 +1,379 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dynamo.Models;
+using DynamoWebServer;
+using DynamoWebServer.Messages;
+using DynamoWebServer.Responses;
+using Moq;
+using NUnit.Framework;
+using SuperSocket.SocketBase.Config;
+
+namespace Dynamo.Tests
+{
+    class SaveUploadTests : DynamoViewModelUnitTest
+    {
+        protected WebServer webServer;
+        protected bool proxyResponseExpected;
+
+        [SetUp]
+        public override void Init()
+        {
+            base.Init();
```

Comment:
This looks good, I just need confirmation that the base `DynamoModel.Start` gets called exactly once per test case (because you also have `[SetUp]` attribute, so I'm not sure how the initialization process goes).